.github/workflows/ci.yml

```diff
@@ -2,137 +2,25 @@ name: CI
 
 on:
   push:
-    branches: [ master ]
+    branches:
+      - master
   pull_request:
 
+
+concurrency:
+  group: ${{ github.ref_name }}-${{ github.workflow }}
+  cancel-in-progress: true
+
 jobs:
   rubocop:
-      runs-on: ubuntu-latest
-      steps:
-        - uses: actions/checkout@v3
-        - name: Setup prerequisites theforeman-rubocop
-          run: |
-            sudo apt-get update && sudo apt-get -y install libyaml-dev
-        - name: Setup Ruby
-          uses: ruby/setup-ruby@v1
-          with:
-            ruby-version: 2.7
-            bundler-cache: true
-        - name: Run rubocop
-          run: bundle exec rubocop -P --format github
+    name: Rubocop
+    uses: theforeman/actions/.github/workflows/rubocop.yml@v0
+    with:
+      command: bundle exec rubocop --parallel --format github
 
-  build:
-    runs-on: ubuntu-latest
+  test:
+    name: Ruby
     needs: rubocop
-
-    services:
-      db:
-        image: postgres:13
-        ports: ['5432:5432']
-        env:
-          POSTGRES_USER: foreman
-          POSTGRES_PASSWORD: foreman
-          POSTGRES_DB: foreman
-        options: >-
-          --health-cmd pg_isready
-          --health-interval 10s
-          --health-timeout 5s
-          --health-retries 5
-    
-    steps:
-    - uses: actions/checkout@v3
-      with:
-        repository: theforeman/foreman
-        ref: develop
-        path: foreman
-    - uses: actions/checkout@v3
-      with:
-        repository: theforeman/foreman_fog_proxmox
-        path: foreman_fog_proxmox
-    - uses: actions/checkout@v3
-      with:
-        repository: fog/fog-proxmox
-        path: fog-proxmox
-    - uses: ruby/setup-ruby@v1
-      with:
-        ruby-version: '2.7'
-        bundler-cache: true
-    - name: Setup foreman
-      env:
-        PGHOST: 127.0.0.1
-        PGDB: foreman
-        PGPWD: foreman
-        PGUSER: foreman
-        RAILS_ENV: test
-        DISABLE_SPRING: true
```

Comment:
We only install `spring` in the `development` group in Foreman (https://github.com/theforeman/foreman/blob/066dd49a9ec121038bd93ac7d03f28ea0fc4b69c/bundler.d/development.rb#L23) but that group is not installed when we run tests, so no need to disable spring